@embhorn please resolve merge conflicts.

FYI: Warnings I see still are:

```
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(516,24): warning C4244: '=': conversion from 'word32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(743,29): warning C4244: 'initializing': conversion from 'const sword32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(744,29): warning C4244: 'initializing': conversion from 'const sword32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(745,29): warning C4244: 'initializing': conversion from 'const sword32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(746,29): warning C4244: 'initializing': conversion from 'const sword32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(747,29): warning C4244: 'initializing': conversion from 'const sword32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(748,29): warning C4244: 'initializing': conversion from 'const sword32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(749,29): warning C4244: 'initializing': conversion from 'const sword32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(750,29): warning C4244: 'initializing': conversion from 'const sword32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1041,43): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1042,35): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1043,28): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1044,35): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1045,35): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1046,28): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1047,35): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1048,28): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1049,35): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1050,35): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1051,28): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1052,35): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1054,28): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1065,42): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1066,33): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1067,33): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1068,33): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1069,27): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1070,42): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1071,33): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1072,33): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1074,33): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1075,27): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1003,65): warning C4244: 'initializing': conversion from 'int' to 'word16', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1004,65): warning C4244: 'initializing': conversion from 'int' to 'word16', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1005,65): warning C4244: 'initializing': conversion from 'int' to 'word16', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1006,65): warning C4244: 'initializing': conversion from 'int' to 'word16', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1007,65): warning C4244: 'initializing': conversion from 'int' to 'word16', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1008,65): warning C4244: 'initializing': conversion from 'int' to 'word16', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1009,65): warning C4244: 'initializing': conversion from 'int' to 'word16', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1010,65): warning C4244: 'initializing': conversion from 'int' to 'word16', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1012,48): warning C4244: 'initializing': conversion from 'int' to 'word16', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1014,48): warning C4244: 'initializing': conversion from 'int' to 'word16', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1016,48): warning C4244: 'initializing': conversion from 'int' to 'word16', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1018,48): warning C4244: 'initializing': conversion from 'int' to 'word16', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1020,48): warning C4244: 'initializing': conversion from 'int' to 'word16', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1022,48): warning C4244: 'initializing': conversion from 'int' to 'word16', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1024,48): warning C4244: 'initializing': conversion from 'int' to 'word16', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1026,48): warning C4244: 'initializing': conversion from 'int' to 'word16', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1339,17): warning C4244: '=': conversion from 'word32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1340,20): warning C4244: '=': conversion from 'word32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1341,27): warning C4244: '=': conversion from 'word32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1342,20): warning C4244: '=': conversion from 'word32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1343,27): warning C4244: '=': conversion from 'word32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1344,20): warning C4244: '=': conversion from 'word32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1345,27): warning C4244: '=': conversion from 'word32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1346,20): warning C4244: '=': conversion from 'word32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1348,20): warning C4244: '=': conversion from 'word32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1391,17): warning C4244: '=': conversion from 'sword32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1392,20): warning C4244: '=': conversion from 'sword32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1393,27): warning C4244: '=': conversion from 'sword32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1394,20): warning C4244: '=': conversion from 'sword32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1395,20): warning C4244: '=': conversion from 'sword32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1396,17): warning C4244: '=': conversion from 'sword32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1397,20): warning C4244: '=': conversion from 'sword32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1398,27): warning C4244: '=': conversion from 'sword32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1399,20): warning C4244: '=': conversion from 'sword32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(1400,20): warning C4244: '=': conversion from 'sword32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(7642,18): warning C4244: 'function': conversion from 'word32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(7740,18): warning C4244: 'function': conversion from 'word32' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(9681,51): warning C4244: 'function': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(9962,59): warning C4244: 'function': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(9530): warning C4701: potentially uninitialized local variable 'inLen' used
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(8991): warning C4701: potentially uninitialized local variable 'params' used
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(8991): warning C4703: potentially uninitialized local pointer variable 'params' used
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(10066): warning C4701: potentially uninitialized local variable 'pubKey' used
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(10066): warning C4703: potentially uninitialized local pointer variable 'pubKey' used
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(7071): warning C4701: potentially uninitialized local variable 'oidMsgHashLen' used
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(7735): warning C4701: potentially uninitialized local variable 'oidMsgHashLen' used
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(2866): warning C4701: potentially uninitialized local variable 's' used
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\dilithium.c(2849): warning C4701: potentially uninitialized local variable 'k' used
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber.c(464,22): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(384,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(385,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(386,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(387,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(397,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(398,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(399,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(400,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(440,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(441,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(442,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(443,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(453,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(454,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(455,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(456,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(466,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(467,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(468,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(469,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(504,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(505,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(506,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(507,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(517,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(518,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(519,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(520,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(810,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(812,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(814,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(816,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(827,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(829,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(831,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(833,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(873,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(875,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(877,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(879,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(890,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(892,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(894,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(896,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(907,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(909,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(911,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(913,18): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(949,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(951,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(953,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(955,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(966,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(968,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(970,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(972,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(983,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(985,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(987,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(989,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(991,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(993,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(995,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(997,14): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(1042,12): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(1045,12): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(1051,12): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(1520,34): warning C4310: cast truncates constant value
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(2951,45): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(2952,45): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(2956,45): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(2957,45): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(4021,69): warning C4244: 'function': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(4029,73): warning C4244: 'function': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(4035,32): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(4195,18): warning C4146: unary minus operator applied to unsigned type, result still unsigned
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(4568,25): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(4569,32): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(4570,32): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(4571,25): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(4572,32): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(4573,32): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(4574,32): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(4575,25): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(4576,32): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(4577,32): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(4578,25): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(5497,32): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(5498,38): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber_poly.c(5499,32): warning C4244: '=': conversion from 'int' to 'byte', possible loss of data
1>C:\msys64\home\DavidGarske\wolfssl\wolfcrypt\src\wc_kyber.c(1221): warning C4701: potentially uninitialized local variable 'compVecSz' used
```

--- a/examples/server/server.c
+++ b/examples/server/server.c
@@ -2382,10 +2382,13 @@ THREAD_RETURN WOLFSSL_THREAD server_test(void* args)
 
 #ifdef HAVE_PQC
             case 259:
+            {
                 usePqc = 1;
+    #if defined(WOLFSSL_TLS13) && defined(HAVE_SUPPORTED_CURVES)
                 onlyKeyShare = 2;
+    #endif
                 pqcAlg = myoptarg;
-                break;
+            } break;
 #endif
 
 #if defined(WOLFSSL_TLS13) && defined(HAVE_SESSION_TICKET)

--- a/src/tls.c
+++ b/src/tls.c
@@ -8168,48 +8168,33 @@ typedef struct PqcHybridMapping {
 
 static const PqcHybridMapping pqc_hybrid_mapping[] = {
 #ifndef WOLFSSL_NO_ML_KEM
-    {.hybrid = WOLFSSL_P256_ML_KEM_512,     .ecc = WOLFSSL_ECC_SECP256R1,
-     .pqc = WOLFSSL_ML_KEM_512,             .pqc_first = 0},
-    {.hybrid = WOLFSSL_P384_ML_KEM_768,     .ecc = WOLFSSL_ECC_SECP384R1,
-     .pqc = WOLFSSL_ML_KEM_768,             .pqc_first = 0},
-    {.hybrid = WOLFSSL_P256_ML_KEM_768,     .ecc = WOLFSSL_ECC_SECP256R1,
-     .pqc = WOLFSSL_ML_KEM_768,             .pqc_first = 0},
-    {.hybrid = WOLFSSL_P521_ML_KEM_1024,    .ecc = WOLFSSL_ECC_SECP521R1,
-     .pqc = WOLFSSL_ML_KEM_1024,            .pqc_first = 0},
-    {.hybrid = WOLFSSL_P384_ML_KEM_1024,    .ecc = WOLFSSL_ECC_SECP384R1,
-     .pqc = WOLFSSL_ML_KEM_1024,            .pqc_first = 0},
+    {WOLFSSL_P256_ML_KEM_512, WOLFSSL_ECC_SECP256R1, WOLFSSL_ML_KEM_512, 0},
+    {WOLFSSL_P384_ML_KEM_768, WOLFSSL_ECC_SECP384R1, WOLFSSL_ML_KEM_768, 0},
+    {WOLFSSL_P256_ML_KEM_768, WOLFSSL_ECC_SECP256R1, WOLFSSL_ML_KEM_768, 0},
+    {WOLFSSL_P521_ML_KEM_1024, WOLFSSL_ECC_SECP521R1, WOLFSSL_ML_KEM_1024, 0},
+    {WOLFSSL_P384_ML_KEM_1024, WOLFSSL_ECC_SECP384R1, WOLFSSL_ML_KEM_1024, 0},
 #ifdef HAVE_CURVE25519
-    {.hybrid = WOLFSSL_X25519_ML_KEM_512,   .ecc = WOLFSSL_ECC_X25519,
-     .pqc = WOLFSSL_ML_KEM_512,             .pqc_first = 1},
-    {.hybrid = WOLFSSL_X25519_ML_KEM_768,   .ecc = WOLFSSL_ECC_X25519,
-     .pqc = WOLFSSL_ML_KEM_768,             .pqc_first = 1},
+    {WOLFSSL_X25519_ML_KEM_512, WOLFSSL_ECC_X25519, WOLFSSL_ML_KEM_512, 1},
+    {WOLFSSL_X25519_ML_KEM_768, WOLFSSL_ECC_X25519, WOLFSSL_ML_KEM_768, 1},
 #endif
 #ifdef HAVE_CURVE448
-    {.hybrid = WOLFSSL_X448_ML_KEM_768,     .ecc = WOLFSSL_ECC_X448,
-     .pqc = WOLFSSL_ML_KEM_768,             .pqc_first = 1},
+    {WOLFSSL_X448_ML_KEM_768, WOLFSSL_ECC_X448, WOLFSSL_ML_KEM_768, 1},
 #endif
 #endif /* WOLFSSL_NO_ML_KEM */
 #ifdef WOLFSSL_MLKEM_KYBER
-    {.hybrid = WOLFSSL_P256_KYBER_LEVEL1,   .ecc = WOLFSSL_ECC_SECP256R1,
-     .pqc = WOLFSSL_KYBER_LEVEL1,           .pqc_first = 0},
-    {.hybrid = WOLFSSL_P384_KYBER_LEVEL3,   .ecc = WOLFSSL_ECC_SECP384R1,
-     .pqc = WOLFSSL_KYBER_LEVEL3,           .pqc_first = 0},
-    {.hybrid = WOLFSSL_P256_KYBER_LEVEL3,   .ecc = WOLFSSL_ECC_SECP256R1,
-     .pqc = WOLFSSL_KYBER_LEVEL3,           .pqc_first = 0},
-    {.hybrid = WOLFSSL_P521_KYBER_LEVEL5,   .ecc = WOLFSSL_ECC_SECP521R1,
-     .pqc = WOLFSSL_KYBER_LEVEL5,           .pqc_first = 0},
+    {WOLFSSL_P256_KYBER_LEVEL1, WOLFSSL_ECC_SECP256R1, WOLFSSL_KYBER_LEVEL1, 0},
+    {WOLFSSL_P384_KYBER_LEVEL3, WOLFSSL_ECC_SECP384R1, WOLFSSL_KYBER_LEVEL3, 0},
+    {WOLFSSL_P256_KYBER_LEVEL3, WOLFSSL_ECC_SECP256R1, WOLFSSL_KYBER_LEVEL3, 0},
+    {WOLFSSL_P521_KYBER_LEVEL5, WOLFSSL_ECC_SECP521R1, WOLFSSL_KYBER_LEVEL5, 0},
 #ifdef HAVE_CURVE25519
-    {.hybrid = WOLFSSL_X25519_KYBER_LEVEL1, .ecc = WOLFSSL_ECC_X25519,
-     .pqc = WOLFSSL_KYBER_LEVEL1,           .pqc_first = 0},
-    {.hybrid = WOLFSSL_X25519_KYBER_LEVEL3, .ecc = WOLFSSL_ECC_X25519,
-     .pqc = WOLFSSL_KYBER_LEVEL3,           .pqc_first = 0},
+    {WOLFSSL_X25519_KYBER_LEVEL1, WOLFSSL_ECC_X25519, WOLFSSL_KYBER_LEVEL1, 0},
+    {WOLFSSL_X25519_KYBER_LEVEL3, WOLFSSL_ECC_X25519, WOLFSSL_KYBER_LEVEL3, 0},
 #endif
 #ifdef HAVE_CURVE448
-    {.hybrid = WOLFSSL_X448_KYBER_LEVEL3,   .ecc = WOLFSSL_ECC_X448,
-     .pqc = WOLFSSL_KYBER_LEVEL3,           .pqc_first = 0},
+    {WOLFSSL_X448_KYBER_LEVEL3, WOLFSSL_ECC_X448, WOLFSSL_KYBER_LEVEL3, 0},
 #endif
 #endif /* WOLFSSL_MLKEM_KYBER */
-    {.hybrid = 0, .ecc = 0, .pqc = 0, .pqc_first = 0}
+    {0, 0, 0, 0}
 };
 
 /* Map an ecc-pqc hybrid group into its ecc group and pqc kem group. */

--- a/wolfcrypt/src/dilithium.c
+++ b/wolfcrypt/src/dilithium.c
@@ -8950,8 +8950,9 @@ int wc_dilithium_check_key(dilithium_key* key)
      */
 
     if (ret == 0) {
-        params = key->params;
         unsigned int allocSz;
+
+        params = key->params;
 
         /* s1-L, s2-K, t0-K, t-K, t1-K */
         allocSz = params->s1Sz + 4 * params->s2Sz;

--- a/wolfcrypt/src/wc_mlkem.c
+++ b/wolfcrypt/src/wc_mlkem.c
@@ -821,36 +821,39 @@ static int mlkemkey_encapsulate(MlKemKey* key, const byte* m, byte* r, byte* c)
     }
     if (ret == 0) {
 #endif
-        byte* c1 = c;
-        byte* c2 = c + compVecSz;
+        {
+            byte* c1 = c;
+            byte* c2 = c + compVecSz;
 
-    #if defined(WOLFSSL_KYBER512) || defined(WOLFSSL_WC_ML_KEM_512)
-        if (k == WC_ML_KEM_512_K) {
-            /* Step 22: c_1 <- ByteEncode_d_u(Compress_d_u(u)) */
-            mlkem_vec_compress_10(c1, u, k);
-            /* Step 23: c_2 <- ByteEncode_d_v(Compress_d_v(v)) */
-            mlkem_compress_4(c2, v);
-            /* Step 24: return c <- (c_1||c_2) */
+        #if defined(WOLFSSL_KYBER512) || defined(WOLFSSL_WC_ML_KEM_512)
+            if (k == WC_ML_KEM_512_K) {
+                /* Step 22: c_1 <- ByteEncode_d_u(Compress_d_u(u)) */
+                mlkem_vec_compress_10(c1, u, k);
+                /* Step 23: c_2 <- ByteEncode_d_v(Compress_d_v(v)) */
+                mlkem_compress_4(c2, v);
+                /* Step 24: return c <- (c_1||c_2) */
+            }
+        #endif
+        #if defined(WOLFSSL_KYBER768) || defined(WOLFSSL_WC_ML_KEM_768)
+            if (k == WC_ML_KEM_768_K) {
+                /* Step 22: c_1 <- ByteEncode_d_u(Compress_d_u(u)) */
+                mlkem_vec_compress_10(c1, u, k);
+                /* Step 23: c_2 <- ByteEncode_d_v(Compress_d_v(v)) */
+                mlkem_compress_4(c2, v);
+                /* Step 24: return c <- (c_1||c_2) */
+            }
+        #endif
+        #if defined(WOLFSSL_KYBER1024) || defined(WOLFSSL_WC_ML_KEM_1024)
+            if (k == WC_ML_KEM_1024_K) {
+                /* Step 22: c_1 <- ByteEncode_d_u(Compress_d_u(u)) */
+                mlkem_vec_compress_11(c1, u);
+                /* Step 23: c_2 <- ByteEncode_d_v(Compress_d_v(v)) */
+                mlkem_compress_5(c2, v);
+                /* Step 24: return c <- (c_1||c_2) */
+            }
+        #endif
+
         }
-    #endif
-    #if defined(WOLFSSL_KYBER768) || defined(WOLFSSL_WC_ML_KEM_768)
-        if (k == WC_ML_KEM_768_K) {
-            /* Step 22: c_1 <- ByteEncode_d_u(Compress_d_u(u)) */
-            mlkem_vec_compress_10(c1, u, k);
-            /* Step 23: c_2 <- ByteEncode_d_v(Compress_d_v(v)) */
-            mlkem_compress_4(c2, v);
-            /* Step 24: return c <- (c_1||c_2) */
-        }
-    #endif
-    #if defined(WOLFSSL_KYBER1024) || defined(WOLFSSL_WC_ML_KEM_1024)
-        if (k == WC_ML_KEM_1024_K) {
-            /* Step 22: c_1 <- ByteEncode_d_u(Compress_d_u(u)) */
-            mlkem_vec_compress_11(c1, u);
-            /* Step 23: c_2 <- ByteEncode_d_v(Compress_d_v(v)) */
-            mlkem_compress_5(c2, v);
-            /* Step 24: return c <- (c_1||c_2) */
-        }
-    #endif
     }
 
 #ifndef WOLFSSL_NO_MALLOC

